types/layer.go

```diff
@@ -125,19 +126,61 @@ func (l *Layer) ParseFullCommand() ([]string, error) {
 	})
 }
 
-func (l *Layer) ParseImport() ([]string, error) {
-	rawImports, err := l.getStringOrStringSlice(l.Import, func(s string) ([]string, error) {
-		return strings.Split(s, "\n"), nil
-	})
-	if err != nil {
-		return nil, err
+// Import directive from stacker file needs to support both []string and []map[string]string eg:
+// Layer.Import: [{"hash": importHash, "path": importPath}] && [importPath1, importPath2]
+// UpdateLayerImports converts Layer.Import from []interface{} to []map[string]string
+// Make sure we call this before reading stacker files and before opening cache
+func (l *Layer) UpdateLayerImports() error {
```

Comment:
Instead of mutating `l` here, can we just have this return the list of parsed imports?